x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/StartsWith.java

```diff
@@ -6,115 +6,20 @@
-    private final Expression pattern;
-
-    public StartsWith(Source source, Expression src, Expression pattern) {
-        super(source, Arrays.asList(src, pattern));
-        this.source = src;
-        this.pattern = pattern;
-    }
-
-    @Override
-    protected TypeResolution resolveType() {
-        if (!childrenResolved()) {
-            return new TypeResolution("Unresolved children");
-        }
-
-        TypeResolution sourceResolution = isStringAndExact(source, sourceText(), ParamOrdinal.FIRST);
-        if (sourceResolution.unresolved()) {
-            return sourceResolution;
-        }
-
-        return isStringAndExact(pattern, sourceText(), ParamOrdinal.SECOND);
+    public StartsWith(Source source, Expression field, Expression pattern, Configuration configuration) {
+        super(source, field, pattern, configuration);
     }
 
     @Override
-    protected Pipe makePipe() {
-        return new StartsWithFunctionPipe(source(), this, Expressions.pipe(source), Expressions.pipe(pattern));
-    }
-
-    @Override
-    public boolean foldable() {
-        return source.foldable() && pattern.foldable();
-    }
-
-    @Override
-    public Object fold() {
-        return doProcess(source.fold(), pattern.fold());
-    }
-
-    @Override
-    protected NodeInfo<? extends Expression> info() {
-        return NodeInfo.create(this, StartsWith::new, source, pattern);
-    }
-
-    @Override
-    public ScriptTemplate asScript() {
-        ScriptTemplate sourceScript = asScript(source);
-        ScriptTemplate patternScript = asScript(pattern);
-
-        return asScriptFrom(sourceScript, patternScript);
-    }
-    
-    protected ScriptTemplate asScriptFrom(ScriptTemplate sourceScript, ScriptTemplate patternScript) {
-        return new ScriptTemplate(format(Locale.ROOT, formatTemplate("{eql}.%s(%s,%s)"),
-                "startsWith",
-                sourceScript.template(),
-                patternScript.template()),
-                paramsBuilder()
-                    .script(sourceScript.params())
-                    .script(patternScript.params())
-                    .build(), dataType());
-    }
-
-    @Override
-    public ScriptTemplate scriptWithField(FieldAttribute field) {
-        return new ScriptTemplate(processScript(Scripts.DOC_VALUE),
-                paramsBuilder().variable(field.exactAttribute().name()).build(),
-                dataType());
-    }
-
-    @Override
-    public DataType dataType() {
-        return DataTypes.BOOLEAN;
-    }
-
-    @Override
-    public Expression replaceChildren(List<Expression> newChildren) {
-        if (newChildren.size() != 2) {
-            throw new IllegalArgumentException("expected [2] children but received [" + newChildren.size() + "]");
-        }
-
-        return new StartsWith(source(), newChildren.get(0), newChildren.get(1));
+    public boolean isCaseSensitive() {
+        return ((EqlConfiguration) configuration()).isCaseSensitive();
```

Comment:
Maybe add isCaseSensitive to the ql configuration with default value and avoid the cast?

Reply:
I disagree - that would cause `SqlConfiguration` to have a case sensitivity option that is meaningless.